src/inputs/internal/ComboBoxBase.tsx

```diff
@@ -306,6 +306,8 @@ export function ComboBoxBase<O, V extends Value>(props: ComboBoxBaseProps<O, V>)
 
   // Reset inputValue when closed or selected changes
   useEffect(() => {
+    // if we are using the onSearch functionality with debounceSearch, we don't want to reset the input value
+    if (onSearch && debouncedSearch) return;
```

Comment:
Fwiw @apattersonATX-HB / @kbesingeryeh we've had enough back/forth/churn on this issue, that it would be amazing to have some unit test coverage in this PR.

Looking back at Brandon's last PR that was futzing with this, it was in ~April 2024 when "beam was dead" meant we skimped on test coverage :grimacing: 

That said, I'm sure this edge case is trickier than usual to get covered/exposed in a unit test -- but maybe you could give it a shot? Disclaimer I haven't looked at the existing combo box/select field tests, and whether we already have some "intricate" tests (i.e. key here, focus there, see the list do x/y/z) to copy/paste as a starting point--hopefully! :crossed_fingers:

Reply:
yes, I can work on adding a test!

Reply:
Thanks for the extra context @stephenh and thanks for the tests @kbesingeryeh !

Beam was dead, but it has risen!!!!

Reply:
@stephenh @apattersonATX-HB I spent some time trying to add a test and it gets tricky as the issue lies in how the internal state of `ComboBoxBase` was resetting the input when we didn't have an early return from the `debounce`. We don't currently have any tests for `ComboBoxBase` - only the components that use it as a base.

I added a story for the search functionality in a `multi-filter` and a test for the search functionality, but I don't think those would have caught this one-off issue with the user typing and internal state resetting. Let me know if this is enough otherwise I may need some additional help

Reply:
Ah I think the test you need to setup involves mocking a gql call that'll update the filter options? Or do a search more? If we don't have that test setup in place now (@stephenh do we?) just go ahead and merge this in and add a follow up PR since the setup might be a pain 🙏🏽